award_emojis.go

```diff
@@ -0,0 +1,328 @@
+	return a.createAwardEmoji(pid, awardSnippets, snippetID, awardID, options...)
+}
+
+func (a *AwardEmojiService) createAwardEmoji(pid interface{}, resource string, resourceID, awardID int, options ...OptionFunc) (*AwardEmoji, *Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, nil, err
+	}
+	u := fmt.Sprintf("projects/%s/%s/%d/award_emoji/%d", url.QueryEscape(project), resource, resourceID, awardID)
+
+	req, err := a.client.NewRequest("POST", u, nil, options)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	awardEmoji := new(AwardEmoji)
+	resp, err := a.client.Do(req, &awardEmoji)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return awardEmoji, resp, err
+}
+
+// DeleteIssueAwardEmoji delete award emoji on an issue
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#award-a-new-emoji-on-a-note
+func (a *AwardEmojiService) DeleteIssueAwardEmoji(pid interface{}, issueIID, awardID int, options ...OptionFunc) (*Response, error) {
+	return a.deleteAwardEmoji(pid, awardMergeRequest, issueIID, awardID, options...)
+}
+
+// DeleteMergeRequestAwardEmoji delete award emoji on a merge request
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#award-a-new-emoji-on-a-note
+func (a *AwardEmojiService) DeleteMergeRequestAwardEmoji(pid interface{}, mergeRequestIID, awardID int, options ...OptionFunc) (*Response, error) {
+	return a.deleteAwardEmoji(pid, awardMergeRequest, mergeRequestIID, awardID, options...)
+}
+
+// DeleteSnippetAwardEmoji delete award emoji on a snippet
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#award-a-new-emoji-on-a-note
+func (a *AwardEmojiService) DeleteSnippetAwardEmoji(pid interface{}, snippetID, awardID int, options ...OptionFunc) (*Response, error) {
+	return a.deleteAwardEmoji(pid, awardMergeRequest, snippetID, awardID, options...)
+}
+
+// DeleteAwardEmoji Delete an award emoji on the specified resource
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#delete-an-award-emoji
+func (a *AwardEmojiService) deleteAwardEmoji(pid interface{}, resource string, resourceID, awardID int, options ...OptionFunc) (*Response, error) {
+	project, err := parseID(pid)
+	if err != nil {
+		return nil, err
+	}
+	u := fmt.Sprintf("projects/%s/%s/%d/award_emoji/%d", url.QueryEscape(project), resource, resourceID, awardID)
+
+	req, err := a.client.NewRequest("DELETE", u, nil, options)
+	if err != nil {
+		return nil, err
+	}
+	return a.client.Do(req, nil)
+}
+
+// ListAwardEmojiOnNote list single note's award emoji
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#list-a-note-39-s-award-emoji
+func (a *AwardEmojiService) ListAwardEmojiOnNote(pid interface{}, issueIID, noteID int, options ...OptionFunc) ([]*AwardEmoji, *Response, error) {
```

Comment:
Can you add these `OnNote` funcs to work like the others (factory func passing in constants for the resource). Its not super obvious in the docs, but it appear the notes portion of the API works the same

Ref: https://docs.gitlab.com/ce/api/award_emoji.html#award-emoji-on-notes

Reply:
Ohh right, I see it. I'm gonna try them to be sure it's a valid endpoint for the other resources.

Reply:
I pushed the new code to support all type of resources

Reply:
I think it's probably the same for `Get` right?

Reply:
Yeah they talk about it being supported on all resources before describing the API calls.

Reply:
@zaquestion There you go all the "factory" methods are implemented I think...